### PR TITLE
build: fix aarch64 build with aarch64-linux-gnu-g++-4.8

### PR DESCRIPTION
**Pull request by alalek**
Related #7317

http://pullrequest.opencv.org/buildbot/builders/master_ARMv8-lin/builds/12

### CODE REVIEW

**Comment by alalek** [comment]
@tomoaki0705 Please take a look on this FP16 fix.

File: modules/core/include/opencv2/core/cvdef.h [hunk 1]
```diff
@@ -307,10 +307,16 @@ enum CpuFeatures {
 #define CV_2PI 6.283185307179586476925286766559
 #define CV_LOG2 0.69314718055994530941723212145818
 
+#if defined (__ARM_FP16_FORMAT_IEEE) || defined (__ARM_FP16_FORMAT_ALTERNATIVE)
```
Comment (tomoaki0705) [inline modules/core/include/opencv2/core/cvdef.h]: ARM alternative format is a special format, so if we are going to use FP16 in OpenCV, I think it's better not to support the ```__ARM_FP16_FORMAT_ALTERNATIVE``` and only general IEEE format

Comment (alalek) [inline modules/core/include/opencv2/core/cvdef.h]: Removed

**Comment by alalek** [comment]
:+1:

**Pull request merged**